opencog/query/InitiateSearchCB.cc

```diff
@@ -149,94 +149,208 @@ void InitiateSearchCB::set_pattern(const Variables& vars,
 //
 
 Handle
-InitiateSearchCB::find_starter(const Handle& h, size_t& depth,
-                                     Handle& startrm, size_t& width)
+InitiateSearchCB::find_starter(const Handle& root, size_t& depth,
+                                     Handle& link, size_t& width)
 {
 	// If its a node, then we are done.
-	Type t = h->get_type();
-	if (_nameserver.isNode(t))
+	Type root_node_type = root->get_type();
+	if (_nameserver.isNode(root_node_type))
 	{
-		if (VARIABLE_NODE != t and GLOB_NODE != t)
+		if (VARIABLE_NODE != root_node_type and GLOB_NODE != root_node_type)
 		{
-			width = h->getIncomingSetSize();
-			startrm = h; // XXX wtf ???
-			return h;
+			depth = 0;
+			width = root->getIncomingSetSize();
+			link = root; // XXX wtf ???
+			return root;
 		}
 		return Handle::UNDEFINED;
 	}
 
 	// If its a link, then find recursively
-	return find_starter_recursive(h, depth, startrm, width);
+	return find_starter_recursive(root, depth, link, width);
 }
 
-Handle
-InitiateSearchCB::find_starter_recursive(const Handle& h, size_t& depth,
-                                         Handle& startrm, size_t& width)
+Handle InitiateSearchCB::find_starter_recursive(const Handle& root, size_t& depth,
+                                                Handle& link, size_t& width)
 {
 	// If its a node, then we are done. Don't modify either depth or
 	// start.
-	Type t = h->get_type();
-	if (_nameserver.isNode(t))
+	Type root_node_type = root->get_type();
+	if (_nameserver.isNode(root_node_type))
 	{
-		if (VARIABLE_NODE != t and GLOB_NODE != t)
+		if (VARIABLE_NODE != root_node_type and GLOB_NODE != root_node_type)
 		{
-			width = h->getIncomingSetSize();
-			return h;
+			depth = 0;
+			width = root->getIncomingSetSize();
+			return root;
 		}
 		return Handle::UNDEFINED;
 	}
 
 	// Ignore all dynamically-evaluatable links up front.
-	if (_dynamic->find(h) != _dynamic->end())
+	if (_dynamic->find(root) != _dynamic->end())
 		return Handle::UNDEFINED;
 
 	// Iterate over all the handles in the outgoing set.
 	// Find the deepest one that contains a constant, and start
 	// the search there.  If there are two at the same depth,
 	// then start with the skinnier one.
-	size_t deepest = depth;
-	startrm = Handle::UNDEFINED;
-	Handle hdeepest(Handle::UNDEFINED);
-	size_t thinnest = SIZE_MAX;
+	depth = 0;
+	link = Handle::UNDEFINED;
+	Handle starter;
+	width = SIZE_MAX;
```

Comment:
I  also object to this name change.  We are explicitly looking for the least-wide atom, and not just some "wide" atom.   If you renamed it from "thinnest" to "least_wide", or "skinniest" that would be OK. But to rename it to a completely opaque, totally meaning-less "width" is just code-obfuscation.

Reply:
It is not only renaming - I have replaced separate variable by using method parameter. I can revert it to make code cleaner.